core/src/main/java/io/grpc/util/RoundRobinLoadBalancer.java

```diff
@@ -109,8 +109,18 @@ public void handleResolvedAddresses(ResolvedAddresses resolvedAddresses) {
       }
     }
 
-    // Create new subchannels for new addresses.
-    for (EquivalentAddressGroup addressGroup : addedAddrs) {
+    for (Map.Entry<EquivalentAddressGroup, EquivalentAddressGroup> latestEntry :
+        latestAddrs.entrySet()) {
+      EquivalentAddressGroup strippedAddressGroup = latestEntry.getKey();
+      EquivalentAddressGroup originalAddressGroup = latestEntry.getValue();
+      Subchannel existingSubchannel = subchannels.get(strippedAddressGroup);
+      if (existingSubchannel != null) {
+        // EAG's Attributes may have changed.
+        existingSubchannel.updateAddresses(Collections.singletonList(originalAddressGroup));
```

Comment:
`updateAddresses` is for pick first, I don't think this is the right thing to do here.

Reply:
Sorry, I you hadn't seen some of the other details. You are right that nothing changes immediately. However, the next connection that the subchannel makes will use the updated Attributes, which is what we are wanting right now.